### PR TITLE
Managing AWS instances with Terraform and AWS Lambda

### PR DESCRIPTION
**Pull request by scarolan**
So far the Lambda functions will:

1.  Create a CSV-formatted list of untagged instances
2.  Post that list once per day to a slack channel

### CODE REVIEW

File: infrastructure-as-code/aws-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Terraforming EC2 lifecycles with AWS Lambda
```
Comment (bensojona) [inline infrastructure-as-code/aws-lambda/README.md]: I might make the filename `aws-lambda` a bit more specific as eventually we'll have many AWS Lambda guides. Maybe `ec2-lifecycles-aws-lambda` or something.

File: infrastructure-as-code/aws-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+This Terraform configuration provides examples of:
+ - AWS Lambda functions for performing various ec2 cleanup tasks
+ - IAM policy required to automate Lambda functions
+ - Cloudwatch roles for scheduling Lambda runs
+ - Simple Email Service
+ - Slack integration
+##
```
Comment (bensojona) [inline infrastructure-as-code/aws-lambda/README.md]: I think this is a first cut, but at some point it'd be great to get the README in the Guide template format --> https://docs.google.com/document/d/17zk1a_azbAp0ph4_D_xh8E-LJ6tIFR_5C967ik1Rj9c/edit#heading=h.efo7z8nz6mqv

File: infrastructure-as-code/aws-lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Terraforming EC2 lifecycles with AWS Lambda
+This Terraform configuration provides examples of:
+ - AWS Lambda functions for performing various ec2 cleanup tasks
```
Comment (bensojona) [inline infrastructure-as-code/aws-lambda/README.md]: Will this guide ever expand beyond EC2 instances? Or is that all that's in scope for now?

File: infrastructure-as-code/aws-lambda/README.md (hunk 1 above)
Comment (scarolan) [reply infrastructure-as-code/aws-lambda/README.md]: Ok, i can do that.

File: infrastructure-as-code/aws-lambda/README.md (hunk 3 above)
Comment (scarolan) [reply infrastructure-as-code/aws-lambda/README.md]: Yes, I've tried to design it in such a way that other resources will be easy to support.

File: infrastructure-as-code/aws-lambda/README.md (hunk 2 above)
Comment (scarolan) [reply infrastructure-as-code/aws-lambda/README.md]: Yep, i'll give it the full treatment.  Just getting my python brain cells firing again.  I haven't written python for nearly 5 years.

File: infrastructure-as-code/aws-lambda/README.md (hunk 3 above)
Comment (bensojona) [reply infrastructure-as-code/aws-lambda/README.md]: Awesome. So re: my previous comment, maybe make it `resource-cleanup-aws-lambda` or something more creative than that :)

File: infrastructure-as-code/aws-lambda/README.md (hunk 2 above)
Comment (bensojona) [reply infrastructure-as-code/aws-lambda/README.md]: After I made this comment it was clear this was an early PR haha, carry on

**Pull request merged**